src/platform/packages/shared/kbn-router-to-openapispec/src/generate_oas.ts

```diff
@@ -42,6 +42,11 @@ export interface GenerateOpenApiDocumentOptions {
   tags?: string[];
   env?: Env;
   filters?: GenerateOpenApiDocumentOptionsFilters;
+  /**
+   * Optional handler to post-process the generated OpenAPI document before returning.
+   * Receives the OAS document and should return the modified document.
+   */
+  postProcess?: (oas: OpenAPIV3.Document) => OpenAPIV3.Document | Promise<OpenAPIV3.Document>;
```

Comment:
Can you provide a bit of context on how this will be used? Where will the call to generate OAS be such that a caller can provide this function?